.github/workflows/rust.yml

```diff
@@ -178,3 +202,32 @@ jobs:
       - name: Mark the job as unsuccessful
         if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
         run: exit 1
+
+  publish-release:
+    name: Publish release
+    runs-on: ubuntu-latest
+    needs:
+      ["linux", "mac", "windows"]
+    if: ${{ !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          merge-multiple: true
+      - id: release
+        if: ${{ github.ref == 'refs/heads/main' && github.event_name == 'push' }}
+        run: |
+          RELEASE_TAG=$(date "+%F-%H-%M")
+          gh release create ${RELEASE_TAG} ./*.tar.gz
+          echo "RELEASE_TAG=${RELEASE_TAG}" >> ${GITHUB_OUTPUT}
+    outputs:
+      release-tag: ${{ steps.release.outputs.RELEASE_TAG }}
+
+  verify-release:
+    name: Verify release
+    needs: publish-release
+    if: ${{ !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') }}
+    uses: ./.github/workflows/release.yml
+    with:
+      release-tag: ${{ needs.publish-release.outputs.release-tag }}
+
```

Comment:
This should be moved above build_result job (so the jobs are ordered by their invocation).